docs/src/callbacks.md

```diff
@@ -0,0 +1,102 @@
+# Callbacks
+Many of the advanced features of Trixi, such as adaptive mesh refinement, are implemented as
+callbacks. A callback is an algorithmic entity that gets passed to the ODE solver and
+is called at specific points during execution to perform certain tasks. Callbacks in Trixi are
+either called after each time step (*step callbacks*) or after each stage of the ODE
+solver (*stage callbacks*).
+
+![callbacks_illustration](https://user-images.githubusercontent.com/65298011/107779585-dc3ba700-6d45-11eb-8087-15afc1cd5080.png)
+
+The advantage of callbacks over hard-coding all features is that it allows to extend Trixi without
+modyfing the internal source code. Trixi provides callbacks for time step
+control, adaptive mesh refinement, I/O and more.
+
+## Step Callbacks
```

Comment:
```suggestion
## Step callbacks
```
Please do not capitalize headings (this applies to all other headings in this file as well)

Reply:
Done in fe783972770345b30689982be2f4a293eeffd6c9.